app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/EnterMovementReferenceNumberController.scala

```diff
@@ -428,93 +431,166 @@ class EnterMovementReferenceNumberController @Inject() (
                   )
                 )
               ),
-            movementReferenceNumber =>
-              movementReferenceNumber.value match {
+            mrnOrEntryNumber => {
+
+              val mrnOrEntryValue  = mrnOrEntryNumber.value.map(_.value).leftMap(_.value).merge
+              val numberHasChanged = fillingOutClaim.draftClaim.movementReferenceNumber
+                .map {
+                  case Right(cachedMrn)        => cachedMrn.value =!= mrnOrEntryValue
+                  case Left(cachedEntryNumber) => cachedEntryNumber.value =!= mrnOrEntryValue
+                  case _                       => false
+                }
+                .getOrElse(true)
+
+              mrnOrEntryNumber.value match {
+
                 case Left(entryNumber) =>
-                  val updatedAnswers = answers.fold(
-                    _ =>
-                      CompleteDuplicateMovementReferenceNumberAnswer(
-                        Some(Left(entryNumber))
-                      ),
-                    complete => complete.copy(duplicateMovementReferenceNumber = Some(Left(entryNumber)))
-                  )
-                  val newDraftClaim  =
-                    fillingOutClaim.draftClaim
-                      .fold(_.copy(duplicateMovementReferenceNumberAnswer = Some(updatedAnswers)))
+                  val cachedMrnExists: Boolean = fillingOutClaim.draftClaim.movementReferenceNumber
+                    .map {
+                      case Right(mrn) => mrn.value.isEmpty
+                      case Left(_)    => true
+                    }
+                    .getOrElse(true)
 
-                  val updatedJourney = fillingOutClaim.copy(draftClaim = newDraftClaim)
+                  cachedMrnExists match {
+                    case false =>
+                      val errorKey = "mrn-not-entry-number"
```

Comment:
You have this almost same 5 lines in the code several times. 
It would make more sense to extract it into a method.